pkg/services/tree/service.go

```diff
@@ -24,6 +24,9 @@ type Service struct {
 	replicationTasks chan replicationTask
 	closeCh          chan struct{}
 	containerCache   containerCache
+
+	syncChan chan struct{}
+	cnrMap   map[cidSDK.ID]struct{}
```

Comment:
What exactly does it contain? Do we need it as a map and not as a slice?

Reply:
> What exactly does it contain?

containers that the service is tracking, all the containers that are stored are considered to be in sync

all the containers that do not include the node would be dropped, all the containers that include the node but are not found in that `map` will be synced, and all the containers that could not be synced successfully will be tried to be synced in the next `Sync...All()`

> Do we need it as a map and not as a slice?

yes, we use a rand access when deciding what exact container should be synced on the `NewEpoch` event